strawberryfields/backends/fockbackend/circuit.py

```diff
@@ -314,7 +315,14 @@ def beamsplitter(self, t, r, phi, mode1, mode2):
         """
         Applies a beamsplitter.
         """
-        self._apply_gate(ops.beamsplitter(t, r, phi, self._trunc), [mode1, mode2])
+        mat = ops.beamsplitter(t, r, phi, self._trunc)
+
+        modes = List()
+        modes.append(mode1)
+        modes.append(mode2)
```

Comment:
How come `List` needs to be imported from `numba.typed`, and the modes constructed in this way? Might be worth adding a couple of comments explaining, for other developers.

Reply:
It's needed for Numba to know the type of the parameter passed to the jitted function. I'll add some comments about it.